messenger/make.py

```diff
@@ -1,112 +1,138 @@
-#!/usr/bin/python
 from __future__ import print_function
+
 import os
-import platform
 import sys
 import shlex
 import shutil
 import subprocess
+import platform
+import glob
+import operator
 
+try:
+    from ConfigParser import ConfigParser
+except ImportError:
+    from configparser import ConfigParser
+
+from . import settings
+
+__all__ = [
+    'get_messenger_dir',
+    'get_matlab_bin',
+    'get_config',
+    'do_build',
+    'build_octave',
+    'build_matlab'
+]
 
 def get_messenger_dir():
-    # Check the system platform first
-    splatform = sys.platform
-
-    if splatform.startswith('linux'):
-        messenger_dir = 'mexa64'
-    elif splatform.startswith('darwin'):
-        messenger_dir = 'mexmaci64'
-    elif splatform.startswith('win32'):
-        if splatform == "win32":
-            # We have a win64 messenger, so we need to figure out if this is 32
-            # or 64 bit Windows:
-            if not platform.machine().endswith('64'):
-                raise ValueError("pymatbridge does not work on win32")
-
-        # We further need to differniate 32 from 64 bit:
-        maxint = sys.maxsize
-        if maxint == 9223372036854775807:
-            messenger_dir = 'mexw64'
-        elif maxint == 2147483647:
-            messenger_dir = 'mexw32'
-    return messenger_dir
-
-
-def get_config():
-    messenger_dir = get_messenger_dir()
-    with open(os.path.join(messenger_dir, 'local.cfg')) as fid:
-        lines = fid.readlines()
+    host, is_64bit = platform.system(), platform.machine().endswith('64')
+    ostype = {
+        'Darwin':  'mexmaci64',
+        'Linux':   'mexa64',
+        'Windows': 'mexw64',
+    }
+    if not is_64bit and host == 'Windows':
+        raise ValueError("pymatbridge does not support 32-bit Windows")
 
-    cfg = {}
-    for line in lines:
-        if '=' not in line:
-            continue
-        name, path = line.split('=')
-        cfg[name.lower()] = path.strip() or '.'
-    return cfg
+    return ostype[host] if is_64bit else 'mexw32'
+
+
+def get_config(host, config='config.ini'):
+
+    cfg = ConfigParser()
+    cfg.read(config)
+
+    return dict(cfg.items(host))
 
 
 def do_build(make_cmd, messenger_exe):
     print('Building %s...' % messenger_exe)
     print(make_cmd)
     messenger_dir = get_messenger_dir()
-    subprocess.check_output(shlex.split(make_cmd), shell=True)
+    subprocess.call(shlex.split(make_cmd), stderr=subprocess.STDOUT)
 
     messenger_loc = os.path.join(messenger_dir, messenger_exe)
 
-    shutil.move(messenger_exe, messenger_loc)
+    shutil.move(messenger_exe, os.path.join('messenger', messenger_loc))
 
-    if os.path.exists('messenger.o'):
+    if os.path.isfile('messenger.o'):
         os.remove('messenger.o')
 
 
-def build_octave():
+def build_octave(static=False):
```

Comment:
I only includes it here so to maintain a consistent api for build_matlab and build_octave. I think im going to eventually refactor this out eventually and choose static vs dynamic linking as the best means of dealing with the zmq headers.

I tried looking into pyzmq source code to hopefully get inspiration on how they deal with this issue, but it looks like they ship their installation with precompiled binaries as well.